### PR TITLE
update server to accept multiple trigger types

### PR DESCRIPTION
**Pull request by schoren**
This PR completes the remaining work to support multiple triggers. This breaks backwards compatibility.

## Changes

- Update openAPI definitions
- Update model JSON encodings for DB persistance

## Fixes

- As a side effect, it normalizes the usage of model mappers for websocket responses

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by schoren** [comment]
@mathnogueira I'm working on fixing the CLI now. The file format doesn't need to be updated, so that's good. We'll probably need to work on the FE branching off this branch, and wait until all is done to merge back to main.

**Review by xoscar — APPROVED** [review]
Looking good 🌚

**Review by mathnogueira — APPROVED** [review]
Good to see our tests are working :D, GJ @schoren and @xoscar

**Pull request merged**
